### PR TITLE
feat(kyverno): upgrade kyverno from v1.11.4 to v1.12.6

### PR DESCRIPTION
**Pull request by luigidematteis**
## Relevant changes
A significant improvement to policies execution has been introduced with Kyverno v1.12.6, using a new custom resource, _GlobalContextEntry_ , which allows to group and cache commonly used resources, so that policies can directly access stored data without repeatedly querying the Kubernetes API. This is especially beneficial for policies applied frequently or across multiple namespaces, reducing processing time and resource overhead.

## ⚠️ Breaking (Potentially)
From Kyverno v1.12.0, policies using long-deprecated or invalid operators _in_ conditions (ex., In and NotIn) will be blocked.
Here is the list of valid operators: https://kyverno.io/docs/writing-policies/preconditions/#operators

## Upgrade procedure
After checking that the policies use valid condition operators, the upgrade procedure using YAML files is straightforward, as it is only required to delete the old resources and create the new ones.

### CODE REVIEW

**Comment by ralgozino** [comment]
Hi @luigidematteis, 

- Is this PR still in draft or is it ready to review?
- Are the policies included in the module using the deprecated operators or are they ok?

thanks!

**Comment by luigidematteis** [comment]
> Hi @luigidematteis,
> 
> * Are the policies included in the module using the deprecated operators or are they ok?

The policies included in the module don't use the deperecated operators and all work as expected.
I've added the note to keep track of the potential breaking changes in policies written for previous versions.

File: katalog/kyverno/core/kustomization.yaml [hunk 1]
```diff
@@ -16,19 +16,22 @@ resources:
...
   - name: ghcr.io/kyverno/reports-controller
     newName: registry.sighup.io/fury/kyverno/reports-controller
-    newTag: v1.11.4
+    newTag: v1.12.6
   - name: bitnami/kubectl
     newName: registry.sighup.io/fury/bitnami/kubectl
     newTag: "1.28.5"
+  - name: ghcr.io/kyverno/kyverno-cli
+    newName: registry.sighup.io/fury/kyverno/kyverno-cli
+    newTag: v1.12.6
```
Comment (ralgozino) [inline katalog/kyverno/core/kustomization.yaml:37]: I don't see this image being used in the manifests, maybe something is missing?

**Review by ralgozino — CHANGES_REQUESTED** [review]
Overall the PR is looking good, just a question before approving

**Review by ralgozino — APPROVED** [review]
lgtm

**Pull request merged**